jni-glue/src/string_chars.rs

```diff
@@ -27,16 +27,35 @@ impl<'env> StringChars<'env> {
         }
     }
 
+    /// Construct a StringChars from an Env + AsRef<str>
+    pub fn from_env_str<S: AsRef<str>>(env: &'env Env, string: S) -> Self {
+        let chars = string.as_ref().encode_utf16().collect::<Vec<_>>();
+        let string = unsafe { env.new_string(
+            chars.as_ptr() as *const jchar,
+            chars.len() as jsize,
+        ) };
+        unsafe { Self::from_env_jstring(env, string) }
+    }
+
+    /// Get an Env + jstring from StringChars
+    pub unsafe fn as_env_jstring(&self) -> (&'env Env, jstring) {
+        (self.env, self.string)
+    }
+
```

Comment:
Moving the encoding implementation would also let this method be dropped.  I'm not necessairly against exposing this information as part of the public API if we have a good need/use case for it, but I'm trying to hold off on exposing implementation details as long as possible on this front.

(As an side, this is 100% sound and wouldn't need to be marked unsafe.)

Reply:
I agree